driver-core/src/main/com/mongodb/event/ConnectionRemovedEvent.java

```diff
@@ -17,26 +17,78 @@
 package com.mongodb.event;
 
 import com.mongodb.connection.ConnectionId;
-
-import static com.mongodb.assertions.Assertions.notNull;
+import org.bson.assertions.Assertions;
 
 /**
  * An event for removing a connection from the pool.
  *
  * @since 3.5
  */
 public final class ConnectionRemovedEvent {
+
+    /**
+     * An enumeration of the reasons a connection could be closed
```

Comment:
Adding a new field is not backwards breaking, so now supporting the "reason" property for this event type.

Reply:
ACK